Minor fixes to CAM, CICE, CLM, and POP Diagnostics

duvivier opened the pull request:
Major changes:
1. CAM - added drop down "solutions" to the notebooks with new screen shots of what their results should look like. 
2. CICE - adding drop down "solutions" to the advanced notebook with screen shots of what the results should look like.
3. POP - fixed files issue and casenames and dataset names. Changed questions formatting to have hints below on how to do the suggestion. 
4. CLM - added drop down "solutions" to the notebook with screen shots of the results.
5. Added kernel information at top of each notebook so they use the correct one.
6. Also did some minor formatting to questions and sections so that the TOC on the HTML populates nicely and it's consistent between components.

I have run CAM and CICE all the way through with no issues. 

I ran POP and have a few specific comments below to be addressed (probably in a later PR). 

I also ran CLM and have a few comments - it did hit the memory limit for me but did keep running. But Jupyter under pressure might not work so nicely. Something others should check anyway.

duvivier commented:
@nusbaume I'm also assigning this to you because I assume come Monday I'm going to be kinda crazy hectic. Feel free to merge it if you feel ready and I'm not available.

duvivier commented:
@gustavo-marques 
I also was able to run and update the POP basics notebook. However there are three places I think you can provide info before we finalize it. Either I can add that to this PR or we can merge it and you can add later.
1. Exercise 1, first question. I don't know how you'd recommend plotting 50m temperature. If you (or Anna) can look this over and provide code in the hint drop down that would be great.
2. Exercise 2, first plot (global map). What I get is not what the dropdown shows. Should we change the figure in the drop down or is something going weird with plotting?
3. Exercise 2, second question at very end. Again we suggest plotting SSH or 50m temperature. Can you provide code for how to do this?

I did add a question in exercise 1 about loading in SSH as well and plotting that. So checking the hint answer would be good.

Thanks!

duvivier commented:
> 

Just clarifying that I fixed the POP files issue. We were missing a `\` somewhere in the path and that caused issues. But I fixed that and also modified the way we read in files a bit to use a casename to be more general and to have the dataset be called ds_pop instead of ds_shf (since it wasn't just surface heat flux being read in).

duvivier commented:
@lawrencepj1 

For CLM I went through the exercises to be sure they worked properly. I added drop down "solutions" of each figure so the students can compare their results with the solutions we think they should have. I did not make this change, but you may want to consider adding drop down solutions to the questions as we have done for other components so the students know if they're answering the questions correctly or not. But if you don't do that, it'll be okay too. :)

It would be good if you could go through and be sure everything works as you expect. Jesse had added a section at the beginning where they read in their own data from the tutorial instead of only using the large ensemble.

gustavo-marques commented on notebooks/diagnostics/pop/basics_pop.ipynb:
```diff
@@ -212,16 +224,18 @@
-    "# You can also choose to examine output from a longer POP2 run here:\n",
-    "#pth = '/glade/scratch/gmarques/archive/gpop.e23.GJRA.TL319/ocn/hist/'\n",
+    "# You can also choose to examine output from a longer POP2 run here by uncommenting both lines below:\n",
+    "#casename = 'gpop.e23.GJRA.TL319'\n",
```
I think we should remove this because the data will be purged in a few months.

duvivier replied:
Done, removed these paths.

gustavo-marques commented:
@duvivier, I sent a PR to your branch that addresses the issues you mentioned here, plus additional minor changes

duvivier commented:
> @duvivier, I sent a PR to your branch that addresses the issues you mentioned here, plus additional minor changes. Once you merge my changes, this PR will be updated automatically.

@gustavo-marques I just merged your PR. thank you!
